theano/compile/builders.py

```diff
@@ -137,6 +137,46 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def c_code(self, node, name, inputs, outputs, sub):
+        order = io_toposort(self.new_inputs, self.new_outputs)
+
+        # assert all inner nodes have c_code()
+        if not (self._op_use_c_code and
+                all([getattr(node.op, '_op_use_c_code', False) and
+                     hasattr(node.op, 'c_code') for node in order])):
```

Comment:
Can you test that this way:

                thunk = node.op.make_c_thunk(node, storage_map, compute_map,
                                             no_recycling)

and provide dummy storage_map and compute_map. Set an empty list for no_recycling.